### PR TITLE
Backport missing cluster health fixes from ES

### PR DESCRIPTION
**Pull request by seut**
## Summary of the changes / Why this improves CrateDB

See commits.

This should solve flaky tests which asserts cluster health and shard allocations by waiting for a green cluster health using `ensureGreen`. E.g. the random failures of the `FilteringAllocationIT` should be solved now, the last commit removes the assertBusy calls in there.

### CODE REVIEW

File: server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java [hunk 1]
```diff
@@ -0,0 +1,585 @@
...
+        logger.info("--> trying to get the indexed document on the first round (before close and shutdown)");
+        execute("select id from test where id = 1");
+        assertThat(response.rowCount(), is(1L));
+
+        logger.info("--> indexing a simple document");
+        execute("insert into test (id) values (2)");
+    }
+
+    @Test
+    @Ignore("Creating a table without any data node will take very long as internally at CrateDB, a table creation" +
```
Comment (mfussenegger) [inline server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java:202]: Could we use the slow annotation here instead, or does it currently fail?

**Review by mfussenegger — COMMENTED** [review]
Left a question, otherwise looks good :+1:

**Comment by mfussenegger** [comment]
There is btw. still a diff

image

We've a `masterOperation` overload that got removed in ES - worth following up on?

**Review by mfussenegger — APPROVED** [review]

**Comment by seut** [comment]
@mfussenegger 

> There is btw. still a diff

I've removed the overload (sorry I've already rebased and eliminated the dedicated fixup). Or did you mean it in a different way?

File: server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java (hunk 1 above)
Comment (seut) [reply server/src/test/java/org/elasticsearch/gateway/GatewayIndexStateIT.java]: Was not sure if it's worth and maybe adds another flakiness. I've added a fixup.

**Review by mfussenegger — APPROVED** [review]

**Comment by mfussenegger** [comment]
>  Or did you mean it in a different way?

No, all good :+1:

**Pull request merged**